ui/client/components/sidePanels/SidePanel.tsx

```diff
@@ -0,0 +1,68 @@
+import React, {PropsWithChildren, useState} from "react"
+import cn from "classnames"
+import panelStyles from "./SidePanel.styl"
+import {ScrollbarsExtended} from "./ScrollbarsExtended"
+import ErrorBoundary from "react-error-boundary"
+import TogglePanel from "../TogglePanel"
+import {useDispatch, useSelector} from "react-redux"
+import {isRightPanelOpened, isLeftPanelOpened} from "../../reducers/selectors/ui"
+import {toggleRightPanel, toggleLeftPanel} from "../../actions/nk/ui/layout"
+
+export function useSidePanelToggle(side: "LEFT" | "RIGHT") {
+  const dispatch = useDispatch()
+  const isOpened = useSelector(side === "RIGHT" ? isRightPanelOpened : isLeftPanelOpened)
+  const onToggle = () => dispatch(side === "RIGHT" ? toggleRightPanel() : toggleLeftPanel())
+  return {isOpened, onToggle}
+}
+
+type Props = {
+  isCollapsed?: boolean,
+  className?: string,
+  innerClassName?: string,
+  onScrollToggle?: (isEnabled: boolean) => void,
+}
+
+function ScrollTogglePanel(props: PropsWithChildren<Props>) {
+  const {children, className, innerClassName, isCollapsed, onScrollToggle} = props
+  return (
+    <div className={cn(panelStyles.collapsible, className, !isCollapsed && panelStyles.isOpened)}>
+      <ScrollbarsExtended onScrollToggle={onScrollToggle}>
+        <ErrorBoundary>
+          <div className={cn(panelStyles.sidePanel, innerClassName)}>
+            {children}
+          </div>
+        </ErrorBoundary>
+      </ScrollbarsExtended>
+    </div>
+  )
+}
+
+export enum PanelSide {
+  Right = "RIGHT",
+  Left = "LEFT",
+}
+
+type SidePanelProps = {
+  side: PanelSide,
+  className?: string,
+}
+
+export function SidePanel(props: PropsWithChildren<SidePanelProps>) {
+  const {children, side, className} = props
+  const {isOpened, onToggle} = useSidePanelToggle(side)
+  const [showToggle, setShowToggle] = useState(true)
+
+  return (
+    <>
+      {!isOpened || showToggle ? <TogglePanel type={side} isOpened={isOpened} onToggle={onToggle}/> : null}
+      <ScrollTogglePanel
+        onScrollToggle={setShowToggle}
+        isCollapsed={!isOpened}
+        className={cn(side, side === PanelSide.Left ? panelStyles.left : panelStyles.right)}
```

Comment:
Maybe explicitly tell cn which side the panel is? Just in case we would like to add new panel location
```
cn(side, {[panelStyles.left]: side === PanelSide.Left,[panelStyles.right]: side === PanelSide.Right})
```